File: src/main/kotlin/com/komdatis/dto/BuildingDto.kt
package com.komdatis.dto

data class BuildingDto(
    val id: Long? = null,
    val firstName: String,
    val lastName: String,
    val address: String,
    val livingSpace: Float,
    val warmth: List<BuildingWarmthDto> = emptyList(),
    val warmWater: List<BuildingWarmWaterDto> = emptyList(),
    val heatedBasement: Boolean,
    val apartments: Short
)

File: src/dto/building-dto.ts
import { BuildingWarmWaterDto } from "./building-warm-water-dto";
import { BuildingWarmthDto } from "./building-warmth-dto";

export interface BuildingDto {
  id?: string;
  firstName: string;
  lastName: string;
  address: string;
  livingSpace: number;
  warmth: BuildingWarmthDto[];
  warmWater: BuildingWarmWaterDto[];
  heatedBasement: boolean;
  apartments: number;
}
